src/azure-cli/azure/cli/command_modules/sql/_params.py

```diff
@@ -1476,29 +1476,133 @@ def _configure_security_policy_storage_params(arg_ctx):
-                   'Valid policy for live database is 7-35 days, valid policy for dropped databases is 0-35 days.')
+        c.argument(
+            'retention_days',
+            options_list=['--retention-days'],
+            required=True,
+            help='New backup short term retention policy in days.'
+            'Valid policy for live database is 7-35 days, valid policy for dropped databases is 0-35 days.')
 
     with self.argument_context('sql midb short-term-retention-policy show') as c:
-        c.argument('deleted_time',
-                   options_list=['--deleted-time'],
-                   help='If specified, shows retention days for a deleted database, instead of an existing database.'
-                   'Must match the deleted time of a deleted database on the source Managed Instance.')
+        c.argument(
+            'deleted_time',
+            options_list=['--deleted-time'],
+            help='If specified, shows retention days for a deleted database, instead of an existing database.'
+            'Must match the deleted time of a deleted database on the source Managed Instance.')
+
+    with self.argument_context('sql midb long-term-retention-policy set') as c:
+        create_args_for_complex_type(
+            c, 'parameters', ManagedDatabase, [
+                'weekly_retention',
+                'monthly_retention',
+                'yearly_retention',
+                'week_of_year'
+            ])
+
+        c.argument('weekly_retention',
+                   help='The Weekly Retention.'
+                   'If just a number is passed instead of an ISO 8601 string, days will be assumed as the units.'
+                   'There is a minimum of 7 days and a maximum of 10 years.')
+
+        c.argument('monthly_retention',
+                   help='The Monthly Retention.'
+                   'If just a number is passed instead of an ISO 8601 string, days will be assumed as the units.'
+                   'There is a minimum of 7 days and a maximum of 10 years.')
+
+        c.argument('yearly_retention',
+                   help='The Yearly Retention.'
+                   'If just a number is passed instead of an ISO 8601 string, days will be assumed as the units.'
+                   'There is a minimum of 7 days and a maximum of 10 years.')
+
+        c.argument('week_of_year',
+                   help='The Week of Year, 1 to 52, to save for the Yearly Retention.')
+
+    with self.argument_context('sql midb long-term-retention-backup') as c:
+        c.argument('location_name',
+                   required=True,
+                   arg_type=get_location_type(self.cli_ctx))
+
+        c.argument('resource_group_name',
+                   required=False,
+                   help='If specified, the resource group that the managed instance/database resource belongs to.')
+
+    with self.argument_context('sql midb long-term-retention-backup show') as c:
```

Comment:
What does the --ids mean in your command?
![image](https://user-images.githubusercontent.com/49508232/77729927-c42e3c80-703a-11ea-907c-677fa2434d37.png)
Typically if you want to use id part in your command, the --ids should be id of backup, and --name should be backup name.

Reply:
How do I remove --ids from resource Id arguments? For this function I'm just using the get operation from python sdk.

Reply:
Doesn't --ids refer to resource ids?